src/App.tsx

```diff
@@ -1,28 +1,27 @@
-import { useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
 
+import LayoutComponent from '@/components/layout/LayoutComponent.component.tsx';
+import PageNotFound from '@/components/pageNotFound/PageNotFound.component.tsx';
 import { useTheme } from '@/context/ThemeContext.tsx';
 
 import AboutPage from './components/about/AboutPage.component.tsx';
-import FooterComponent from './components/footer/Footer.component.tsx';
-import HeaderComponent from './components/header/Header.component.tsx';
 import ProductsList from './components/productList/ProductList.component.tsx';
 
 import styles from './App.module.css';
 
 function App() {
-    const [currentComponent, setCurrentComponent] = useState<'About' | 'ProductsList'>('About');
     const { theme } = useTheme();
 
-    const handleChangePage = (component: 'About' | 'ProductsList') => {
-        setCurrentComponent(component);
-    };
-
     return (
         <>
             <div className={`${theme === 'dark' ? styles.darkTheme : styles.lightTheme}`}>
-                <HeaderComponent handleChangePage={handleChangePage} />
-                {currentComponent === 'About' ? <AboutPage /> : <ProductsList />}
-                <FooterComponent />
+                <Routes>
```

Comment:
Create a constant object or enum `ROUTES` to store all page paths there.
You can create a separate component `Router` and keep these route declarations there
Optionally, review the newer way to declare routes https://reactrouter.com/en/main/routers/create-hash-router